pkg/cli/cves_sub_cmd.go

```diff
@@ -0,0 +1,205 @@
+				return err
+			}
+
+			if !strings.HasPrefix(args[0], "CVE") {
+				return fmt.Errorf("%w: expected a cveid 'CVE-...' got '%s'", zerr.ErrInvalidCLIParameter, args[0])
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			searchConfig, err := GetSearchConfigFromFlags(cmd, searchService)
+			if err != nil {
+				return err
+			}
+
+			err = CheckExtEndPointQuery(searchConfig, ImageListForCVEQuery())
+			if err != nil {
+				return fmt.Errorf("%w: '%s'", err, ImageListForCVEQuery().Name)
+			}
+
+			searchedCVEID := args[0]
+			ctx, cancel := context.WithCancel(context.Background())
+			retryCount := 1
+
+			return RetryWithContext(ctx, func() error {
+				err := SearchImagesByCVEIDGQL(searchConfig, repo, searchedCVEID)
+				if err != nil {
+					if !strings.Contains(err.Error(), zerr.ErrCVEDBNotFound.Error()) {
+						cancel()
+
+						return err
+					}
+
+					fmt.Fprintln(searchConfig.resultWriter,
+						"[warning] CVE DB is not ready [", retryCount, "] - retry in ", cveDBRetryInterval, " seconds")
+
+					retryCount += 1
+				}
+
+				return err
+			}, maxRetries, cveDBRetryInterval*time.Second)
+		},
+	}
+
+	imagesByCVEIDCmd.Flags().StringVar(&repo, "repo", "", "Search for a specific CVE by name/id")
+
+	return imagesByCVEIDCmd
+}
+
+func NewFixedTagsCommand(searchService SearchService) *cobra.Command {
+	fixedTagsCmd := &cobra.Command{
+		Use:   "fixed [repo] [cveId]",
+		Short: "List tags which have fixed a CVE",
+		Long:  `List tags which have fixed a CVE`,
+		Args: func(cmd *cobra.Command, args []string) error {
+			const argCount = 2
+
+			if err := cobra.ExactArgs(argCount)(cmd, args); err != nil {
+				return err
+			}
+
+			if !zcommon.CheckIsRepoName(args[0]) {
+				return fmt.Errorf("%w: expected a valid repo name for first argument '%s'", zerr.ErrInvalidCLIParameter, args[0])
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			searchConfig, err := GetSearchConfigFromFlags(cmd, searchService)
+			if err != nil {
+				return err
+			}
+
+			err = CheckExtEndPointQuery(searchConfig, ImageListWithCVEFixedQuery())
+			if err != nil {
+				return fmt.Errorf("%w: '%s'", err, ImageListWithCVEFixedQuery().Name)
+			}
+
+			repo := args[0]
+			searchedCVEID := args[1]
+			ctx, cancel := context.WithCancel(context.Background())
+			retryCount := 1
+
+			return RetryWithContext(ctx, func() error {
+				err := SearchFixedTagsGQL(searchConfig, repo, searchedCVEID)
+				if err != nil {
+					if !strings.Contains(err.Error(), zerr.ErrCVEDBNotFound.Error()) {
+						cancel()
+
+						return err
+					}
+
+					fmt.Fprintln(searchConfig.resultWriter,
+						"[warning] CVE DB is not ready [", retryCount, "] - retry in ", cveDBRetryInterval, " seconds")
+
+					retryCount += 1
+				}
+
+				return err
+			}, maxRetries, cveDBRetryInterval*time.Second)
+		},
+	}
+
+	return fixedTagsCmd
+}
+
+func RetryWithContext(ctx context.Context, operation func() error, maxRetries int,
```

Comment:
It probably is best to move this into a library so that anyone can use this to do a Retry operation.

Reply:
Yes, I agree. There are many useful pieces of code the inside `cli` package that can be moved into their own packages but that we can't yet easily factor out. When we'll drop support for the old command format (using the flags and the searchers in the backend) it will be possible to move them.

Reply:
https://github.com/project-zot/zot/issues/1741